Do not collect network URLs by default and ask if they want to send t…

canaltinova opened the pull request:
This PR fixes [bug 1460727](https://bugzilla.mozilla.org/show_bug.cgi?id=1460727).

julienw commented on src/components/app/ProfileSharing.js:
```diff
@@ -248,6 +256,12 @@ class ProfileSharingCompositeButton extends PureComponent<
 
+  _toggleShareNetworkUrls() {
+    this.setState(prevState => ({
+      shareNetworkUrls: !prevState.shareNetworkUrls,
```
Can we use the checkbox status instead of toggling the state ?
eg:
```javascript
// (notice the change to the method name too)
_onChangeShareNetworkUrls(event: SyntheticEvent<HTMLInputElement>) {
  this.setState({
    shareNetworkUrls: event.currentTarget.checked,
  });
}
```
I find it more "direct". What do you think?

canaltinova replied:
Yeah, this looks better. Thanks!

julienw commented on src/profile-logic/process-profile.js:
```diff
@@ -835,14 +835,35 @@ export function processProfile(
       delete newThread.stringTable;
-      newThread.stringArray = stringTable.serializeToArray();
+      if (includeNetworkUrls === false) {
+        for (let i = 0; i < newThread.markers.data.length; i++) {
```
nit: you can use `newThread.markers.length`, see https://github.com/devtools-html/perf.html/blob/3d02369e79cb77640db96889b762074fedbfa240/src/types/profile.js#L71

julienw commented on src/profile-logic/process-profile.js:
```diff
@@ -835,14 +835,35 @@ export function processProfile(
+            // Strip the URL from the marker name
+            stringArray[newThread.markers.name[i]] = stringArray[
+              newThread.markers.name[i]
+            ].split(':')[0];
```
Can you slice using `indexOf(':')` or a regexp instead ? The goal is to avoid instancianting an array. Also please use a temporary variable for the string index: For example:
```javascript
const stringIndex = newThread.markers.name[i];
stringArray[stringIndex] = stringArray[stringIndex].replace(/:.*/, '');
```
Or:
```javascript
const stringIndex = newThread.markers.name[i];
stringArray[stringIndex] = stringArray[stringIndex].slice(0, stringArray[stringIndex].indexOf(':'));
```
(I prefer the conciseness of the first proposal :) )

canaltinova replied:
Yeah, the first one looks better. Thanks for the suggestion.

julienw commented on src/components/app/ProfileSharing.js:
```diff
@@ -277,6 +291,16 @@ class ProfileSharingCompositeButton extends PureComponent<
+                  />
+                  Share the URLs of all network requests
+                </label>
+              </p>
```
We could use some CSS to center the text with the checkbox:
![screen shot 2018-06-22 at 14 27 04](https://user-images.githubusercontent.com/454175/41776535-6904e5d8-7628-11e8-9edb-a026386e6a42.png)

julienw commented on src/components/app/ProfileSharing.js:
```diff
@@ -42,15 +42,18 @@ const PrivacyNotice = () => (
     >{`To view all the information you can download the full profile to a file and open the
       json structure with a text editor.`}</p>
+    <p>
+      {`The URLs of all network requests will be removed while sharing the profile but keeping the
```
Maybe add "By default" at the start of the sentence ?

canaltinova commented:
Thanks for the review @julienw! Pushed the commit again with the new changes

I'm not sure if we want to keep the share button there. If I would be the user, after sharing the profile, seeing the share button there again would make me think that I didn't share it correctly or something happened during sharing and it failed. Maybe we can write something different to button label to solve this?

canaltinova commented:
@julienw Addressed your review comments and added some space around checkbox to make it a bit more visible. I think we can land this and I can work on adding new "Share with/without URLs" button.

julienw commented on src/components/app/ProfileSharing.css:
```diff
@@ -110,3 +110,13 @@
 }
+
+.profileSharingShareNetworkUrlsContainer {
+  margin: 20px 0;
```
maybe add `-moz-user-select: none; user-select: none` to avoid the text selection but clicking repeatedly on the label.

canaltinova replied:
Added, thanks.

julienw commented on src/components/app/ProfileSharing.css:
```diff
@@ -110,3 +110,13 @@
+.profileSharingShareNetworkUrlsCheckbox {
+  vertical-align: middle;
+  position: relative;
+  bottom: 1px;
```
I tried to find a better solution, using flex, or other things, but couldn't find anything really good, so let's keep this even if I'm not very happy with this "bottom" adjustment...

canaltinova replied:
I'm not very happy with the solution either but I also couldn't find a better one :(

canaltinova commented on src/components/app/ProfileSharing.js:
```diff
@@ -42,15 +42,18 @@ const PrivacyNotice = () => (
+      {`By default, the URLs of all network requests will be removed while sharing the profile
+        but keeping the URLs may help to identify the problems. Please select the checkbox
+        below to share the URLs of the network requests:`}
+    </p>
```
I also didn't understand but I didn't want to change the standard there :)
